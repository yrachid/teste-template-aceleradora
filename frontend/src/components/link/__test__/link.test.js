import React from 'react';
import { render } from '@testing-library/react';
import Link from '../';

const getRegexValidation = (prop) => {
  const message = `Failed prop type: The prop \`${prop}\` is marked as required`;
  const regex = new RegExp(`(.*?)${message}(.*?)`);
  return { message, regex };
};

describe('Link', () => {
  const linkText = 'fake-link';
  const linkProps = {
    className: 'fake-class',
    href: 'http://fake-url.com',
    target: '_blank',
    rel: 'fake-rel'
  };
  let consoleError;

  beforeEach(() => {
    consoleError = jest.spyOn(console, 'error');
  });

  afterEach(() => {
    consoleError.mockRestore();
  });

  test('Should renders component when there is all required props', () => {
    const { container, getByText } = render(<Link {...linkProps}>{linkText}</Link>);

    expect(getByText(linkText)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('Should renders component with warning when there is no children', () => {
    const { regex } = getRegexValidation('children');
    render(<Link {...linkProps} />);

    expect(consoleError).toHaveBeenCalledWith(expect.stringMatching(regex));
  });

  test('Should renders component with warning when there is no href', () => {
    const { regex } = getRegexValidation('href');
    const linkPropsWithoutHref = {...linkProps, href: null};
    render(<Link {...linkPropsWithoutHref}>{linkText}</Link>);

    expect(consoleError).toHaveBeenCalledWith(expect.stringMatching(regex));
  });

  test('Should renders component with warning when there is no target', () => {
    const { regex } = getRegexValidation('target');
    const linkPropsWithoutHref = {...linkProps, target: null};
    render(<Link {...linkPropsWithoutHref}>{linkText}</Link>);

    expect(consoleError).toHaveBeenCalledWith(expect.stringMatching(regex));
  });

  test('Should renders component with warning when there is no rel', () => {
    const { regex } = getRegexValidation('rel');
    const linkPropsWithoutHref = {...linkProps, rel: null};
    render(<Link {...linkPropsWithoutHref}>{linkText}</Link>);

    expect(consoleError).toHaveBeenCalledWith(expect.stringMatching(regex));
  });
});
