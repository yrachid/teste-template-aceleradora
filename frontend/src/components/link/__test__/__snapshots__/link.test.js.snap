// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Should renders component when there is all required props 1`] = `
<div>
  <a
    class="fake-class"
    href="http://fake-url.com"
    rel="fake-rel"
    target="_blank"
  >
    fake-link
  </a>
</div>
`;
